### PR TITLE
Keep "mse" as the metric name in the log

### PR DESCRIPTION
**Pull request by haifeng-jin**
The name `"mse"` was not kept as the name as the metric, but used `"mean_square_error"` instead, when `model.compile(..., metrics=["mse"])`.

The cause of the bug is custom names were only assigned when creating `MeanMetricWrapper` from `Metric`, but some metrics were created directly by deserialization as a `MeanMetricWrapper`.

The solution is to do the name assignment for all metrics before returning.

### CODE REVIEW

File: keras_core/trainers/compile_utils.py [hunk 1]
```diff
@@ -79,13 +79,14 @@ def get_metric(identifier, y_true, y_pred):
...
-        if isinstance(identifier, str):
-            metric_name = identifier
-        else:
-            metric_name = get_object_name(metric_obj)
-        metric_obj = metrics_module.MeanMetricWrapper(
-            metric_obj, name=metric_name
-        )
+        metric_obj = metrics_module.MeanMetricWrapper(metric_obj)
+
+    if isinstance(identifier, str):
```
Comment (fchollet) [inline keras_core/trainers/compile_utils.py:84]:
```
if isinstance(identifier, str):
    metric_obj.name = identifier
```

**Review by fchollet — APPROVED** [review]
LGTM
